core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -94,6 +94,22 @@ private[scheduler] class BlacklistTracker (
   val nodeToBlacklistedExecs = new HashMap[String, HashSet[String]]()
 
   /**
+   * Blacklists permanently the nodes listed in spark.blacklist.alwaysBlacklistedNodes
+   * The blacklist timeout is set to a large value, effectively never expiring.
+   */
+  private val permanentlyBlacklistedNodes: Seq[String] =
+    conf.get("spark.blacklist.alwaysBlacklistedNodes", "").split(',').map(_.trim).filter(_ != "")
+  if (permanentlyBlacklistedNodes.nonEmpty) {
```

Comment:
We should make this a private function and init the val `_nodeBlacklist` with this.